.github/workflows/Release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - "Project.toml"
+jobs:
+  branch:
+    name: "Create Release Branch"
+    runs-on: ubuntu-20.04-16core
+    strategy:
+      fail-fast: false
+    steps:
+        - uses: actions/checkout@v3
+        - uses: julia-actions/setup-julia@v1
+          with:
+            version: "1.8"
+            arch: "x64"
+        - uses: julia-actions/cache@v1
+        - name: Check Project version
+          id: version_check
+          shell: julia --color=yes --project {0}
+          run: |
+            using Pkg.Types
+            project = read_project("Project.toml")
+            latest_tag = parse(VersionNumber, readchomp(`git describe --tags --abbrev=0`))
+            @assert project.version > latest_tag "$(project.version) !> $latest_tag"
+            open(ENV["GITHUB_OUTPUT"], "a") do io
+              println(io, "version=$(project.version)")
+            end
+          continue-on-error: true
+        - name: Checkout new branch
+          if: steps.version_check.outcome = 'success'
+          run: |
+            release_branch="release-v${{ steps.version_check.outputs.version }}"
+            # Error if branch already exists
+            if [[ ! -z $(git ls-remote --heads origin $release_branch) ]]; then
+              echo "Release branch already exists on remote: $release_branch" >&2
+              exit 1
+            fi
+            git checkout -b $release_branch
+        - name: Update Artifacts.toml
+          shell: julia --color=yes --project {0}
+          run: |
+            using Pkg
+            Pkg.instantiate()
+            include("download_tarballs.jl")  # pulls artifacts from the latest CI job
```

Comment:
What's causing this workflow to wait for the CI workflow? At the moment these two workflows run in parallel which and this workflow would always fail. You can either do:
```yaml
on:
  workflow_run:
    workflows:
      - CI
    types: [completed]
    branches: [main]
```
And then perform the `Project.toml` filtering as part of the steps. Or you can include this as a extra job in CI and use `needs`:
```yaml
steps:
  test:
    ...
  
  release:
    needs: [test]
    ...
```
The later option also can utilize `actions/download-artifact` in the `release` job which would allow us to drop `download_tarballs.jl` entirely. An example of how that would work: https://github.com/orgs/community/discussions/26639#discussioncomment-3838014